java-function/accelerator.yaml

```diff
@@ -0,0 +1,148 @@
+      text: HTTP
+    - value: cloudevents
+      text: CloudEvents
+    defaultValue: http
+    required: true
+
+  - name: includeTap
+    label: "Include TAP deployment resources?"
+    inputType: checkbox
+    dataType: boolean
+    defaultValue: true
+    
+  - name: gitUrl
+    label: Git Repository URL
+    description: Change this to the Git repository you will use to store this function's files
+    inputType: text
+    dependsOn:
+      name: includeTap
+    dataType: string
+    required: true
+    defaultValue: "https://github.com/sample-accelerators/java-functions-accelerator.git"
+    
+  - name: gitBranch
+    label: Git Repository Branch
+    description: The corresponding branch of the Git repository
+    inputType: text
+    dependsOn:
+      name: includeTap
+    dataType: string
+    required: true
+    defaultValue: "main"
+
+engine:
+  merge:
+    # Core files
+    - include: [ "src/**", "func.yaml", "LICENSE", "NOTICE", ".gitignore" ]
+
+    # README files
+    - include: [ "README.md", "DEPLOYING.md" ]
+      chain:
+      - type: ReplaceText
+        substitutions:
+        - text: java-function
+          with: "#artifactId"
+
+    # Maven files
+    - include: [ ".mvn/**", "mvnw", "mvnw.bat" ]
+      condition: "#buildType == 'maven'"
+    - include: [ "pom.xml", "jar-assembly.xml" ]
+      condition: "#buildType == 'maven' && #interfaceType == 'http'"
+      chain:
+      - type: ReplaceText
+        substitutions:
+        - text: java-function
+          with: "#artifactId"
+    - include: [ "cloudevents/pom.xml", "cloudevents/jar-assembly.xml" ]
+      condition: "#buildType == 'maven' && #interfaceType == 'cloudevents'"
+      chain:
+      - type: ReplaceText
+        substitutions:
+        - text: java-function
+          with: "#artifactId"
+      - type: RewritePath
+        rewriteTo: "#filename"
+
+    # Gradle files
+    - include: [ "gradle/**", "gradlew", "gradlew.bat" ]
+      condition: "#buildType == 'gradle'"
+    - include: [ "settings.gradle" ]
+      condition: "#buildType == 'gradle'"
+      chain:
+      - type: ReplaceText
+        substitutions:
+        - text: java-function
+          with: "#artifactId"
+    - include: [ "build.gradle" ]
+      condition: "#buildType == 'gradle' && #interfaceType == 'http'"
+    - include: [ "cloudevents/build.gradle" ]
+      condition: "#buildType == 'gradle' && #interfaceType == 'cloudevents'"
+      chain:
+      - type: RewritePath
+        rewriteTo: "#filename"
+
+    # Replace HTTP with CloudEvents
+    - include: [ "cloudevents/Handler.java" ]
+      condition: "#interfaceType == 'cloudevents'"
+      chain:
+      - type: RewritePath
+        rewriteTo: "'src/main/java/functions/Handler.java'"
+
+    # TAP files
+    - include: [ "config/workload.yaml" ]
+      condition: "#includeTap"
+      chain:
+        - type: ReplaceText
+          substitutions:
+            - text: https://github.com/sample-accelerators/java-functions-accelerator.git
```

Comment:
https://github.com/bbtong/application-accelerator-samples/blob/9de4ef965ce0cbecd963c001cd4e858c5f940f57/java-rest-service/config/workload-api-provider.yaml#L35

We noticed that another existing accelerator does this exact same thing.

I'll adjust these git URLs in this PR, but we left them in because of that precedent.